### ISSUE TITLE
Expand type alias for completion provider

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pylance-release#562]
## Environment data

-   Language Server version: 2020.10.3
-   OS and version: macOS
-   Python version: 3.9

## Actual behaviour

Given the code snippet below, Pylance will only suggest `AliasT` as type hint for `var`. `dict[int, str]` is hidden.

## Expected/Suggested behaviour

I often find it useful to directly see the type I'm working with. The type alias is mainly used to keep the code readable and avoid small mistakes when repeatedly writing the type. It might be an idea to display the final type information with the type hit. Some ideas:

```
var: AliasT -> dict[int, str]

var: AliasT = dict[int, str]

var: AliasT
     dict[int, str]
```

## Code Snippet / Additional information

```python
AliasT = dict[int, str]
var: AliasT
```

![Screen Shot 2020-11-04 at 22 31 56](https://user-images.githubusercontent.com/30130371/98170023-945ae780-1eed-11eb-8a89-4345f785c505.png)

**Comment by jakebailey** [comment]
We've put a lot of effort into making sure we can preserve the names all the way up to the tooltips, as the names are often more readable than the full expansion (especially for big union types like numpy's `ArrayLike`, or functions with many many args like in pandas).

I can't really see us wanting to undo this, at least not by default.

**Comment by erictraut** [comment]
If you hover over the alias symbol (e.g. `AliasT` in your example above), you will see the expanded definition. If you hover over some other symbol that refers to the alias (e.g. `var` in your example above), we display the alias rather than expanding it. This behavior is based on strong feedback we received from many users.

**Comment by cdce8p** [comment]
Ok. I see it now. I haven't really used hover before.

Just one more observation. Wouldn't it not at least make sense to expand the type alias itself e.g. while writing function arguments?

![Screen Shot 2020-11-04 at 23 01 38](https://user-images.githubusercontent.com/30130371/98172990-9d9a8300-1ef2-11eb-9902-ee5f5771889f.png)

**Comment by jakebailey** [comment]
That's a good point, I do agree that `AliasT: AliasT` seems a bit suspect.

**Comment by erictraut** [comment]
Yes, good point. The completion provider logic should have the same logic as the hover text provider with respect to type aliases.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.11.2, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#2020112-18-november-2020
